src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -597,7 +598,14 @@ private void WriteBaseFields(BuildEventArgs e, BuildEventArgsFieldFlags flags)
         {
             if ((flags & BuildEventArgsFieldFlags.Message) != 0)
             {
-                WriteDeduplicatedString(e.RawMessage);
+                if (e is BuildCheckResultWarning)
```

Comment:
The other result types will have similar problem.
But let's try to think if we can solve this without special-casing - we'll probably need to to design the BuildCheck results events more properly

Reply:
done in this PR https://github.com/dotnet/msbuild/pull/10265